qmxgraph/widget.py

```diff
@@ -194,22 +204,32 @@ def set_events_bridge(self, bridge):
 
         :param EventsBridge bridge: Bridge with event handlers.
         """
-        self._events_bridge = bridge
+        signal_name_list = (
+            'on_cells_added', 'on_cells_removed', 'on_label_changed',
+            'on_selection_changed', 'on_terminal_changed',
+            'on_terminal_with_port_changed', 'on_view_update',
+        )
+        for signal_name in signal_name_list:
+            own_signal = getattr(self._events_bridge, signal_name)
+            outside_signal = getattr(bridge, signal_name)
+            try:
+                own_signal.disconnect(outside_signal)
+            except TypeError:
+                pass
+            own_signal.connect(outside_signal)
+
+    def _connect_events_bridge(self):
         if self.is_loaded():
-            self._web_view.add_to_js_window('bridge_events_handler', bridge)
-
-            # Bind all known Python/Qt event handlers to JavaScript events
-            self.api.on_cells_added('bridge_events_handler.on_cells_added')
-            self.api.on_cells_removed('bridge_events_handler.on_cells_removed')
-            self.api.on_label_changed('bridge_events_handler.on_label_changed')
+            self.api.on_cells_added('bridge_events_handler.OnCellsAdded')
+            self.api.on_cells_removed('bridge_events_handler.OnCellsRemoved')
+            self.api.on_label_changed('bridge_events_handler.OnLabelChanged')
             self.api.on_selection_changed(
-                'bridge_events_handler.on_selection_changed')
+                'bridge_events_handler.OnSelectionChanged')
```

Comment:
Why change the naming convention here?

Reply:
- `on_selection_changed` is a signal;
- `OnSelectionChanged` is a slot;

With `QWebKit` we could expose `pyqtSinal`s to javascript, they where mapped to functions in javascript.
With `QwebEngine` signal `pyqtSinal`s are exposed as non callable objects. To retain the qmxgraph's python api I am exposing the slots (they are functiuons in javascript) and they just emit the signals.

I added the async "part" in there: the slot asynchronously emits the signal.

I use a different convention because both coexist in the same object. The slots are not part of the public api just internaly to "bridge" the javascript event handlers to python pyqtSinals.

Instead of `OnSelectionChanged` should I use `onSelectionChanged`, `on_selection_changed_slot`, `async_emit_on_selection_changed`, ...?

Reply:
The rest of the convention is always `lower_case`, that's why I asked why you changed to `UpperCase` here, which is a bit strange.

I suggest to use `selection_changed_slot`.